interactive_engine/executor/ir/runtime/src/process/operator/map/expand_intersect.rs

```diff
@@ -13,39 +13,45 @@
 //! See the License for the specific language governing permissions and
 //! limitations under the License.
 
+use std::any::Any;
 use std::collections::BTreeMap;
 use std::convert::TryInto;
-use std::sync::Arc;
 
+use dyn_type::BorrowObject;
 use graph_proxy::apis::graph::element::GraphElement;
-use graph_proxy::apis::{Direction, DynDetails, QueryParams, Statement, Vertex, ID};
+use graph_proxy::apis::{Direction, Element, QueryParams, Statement, ID};
 use ir_common::error::ParsePbError;
 use ir_common::generated::algebra as algebra_pb;
 use ir_common::KeyId;
 use pegasus::api::function::{FilterMapFunction, FnResult};
 use pegasus::codec::{Decode, Encode, ReadExt, WriteExt};
+use pegasus_common::downcast::*;
+use pegasus_common::impl_as_any;
 
 use crate::error::{FnExecError, FnGenError, FnGenResult};
+use crate::process::entry::{DynEntry, Entry, EntryType};
 use crate::process::operator::map::FilterMapFuncGen;
-use crate::process::record::{Entry, Record};
+use crate::process::record::Record;
 
 /// An ExpandOrIntersect operator to expand neighbors
 /// and intersect with the ones of the same tag found previously (if exists).
 /// Notice that edge_or_end_v_tag (the alias of expanded neighbors) must be specified.
-struct ExpandOrIntersect<E: Into<Entry>> {
+struct ExpandOrIntersect<E: Entry> {
     start_v_tag: Option<KeyId>,
     edge_or_end_v_tag: KeyId,
     stmt: Box<dyn Statement<ID, E>>,
 }
 
 #[derive(Debug, Clone, Hash, PartialEq, PartialOrd)]
-pub struct Intersection {
-    vertex_vec: Vec<Vertex>,
+pub struct IntersectionEntry {
+    vertex_vec: Vec<ID>,
```

Comment:
Also comment in details for what the `IntersectionEntry` is, and how it will be used.